src/formatter.cpp

```diff
@@ -3,6 +3,22 @@
 
 #include <RcppSpdlog>
 
+#if defined(SPDLOG_USE_STD_FORMAT) && __cplusplus >= 202002L
+#define NAMESPACE std
+#else
+#define NAMESPACE fmt
+#endif
```

Comment:
One can do this with a typedef too.  

The combination of the C++20 test and the *USE_STD_FORMAT only here is ... a little odd.  Maybe we should check if *USE_STD_FORMAT is set and error if it is not C++20 as well?   Or check in all other uses?

Reply:
Yes erroring out if not C++20 makes more sense